addons/editor-number-arrow-keys/userscript.js

```diff
@@ -105,6 +106,39 @@ export default async function ({ addon }) {
       shiftDecimalPointToRight(e.target.value, 5) + shiftDecimalPointToRight(changeBy.toString(), 5);
     const newValue = shiftDecimalPointToLeft(newValueAsInt.toString(), 5);
 
-    e.target.value = newValue.toString();
+    if (e.target.className.includes("input_input-form_")) {
+      Object.getOwnPropertyDescriptor(e.target.constructor.prototype, "value").set.call(e.target, newValue.toString());
+      e.target.dispatchEvent(new Event("input", { bubbles: true }));
+
+      // The user probably wants to visualize how the sprite changes size, coordinates, direction, etc.
+      // without having to press Enter. But updating in realtime could be bad for performance.
+      const FLUSH_AFTER_MS = 300; // Number of ms to wait until the input change takes effect.
+      // Force flush after incrementing/decrementing 5 times in a row, even if it's not been 300ms:
+      const FLUSH_AFTER_N_STEPS = 5;
+
+      // https://github.com/scratchfoundation/scratch-gui/blob/develop/src/components/forms/buffered-input-hoc.jsx
+      // This function calls handleFlush() on the buffered input when called.
+      const flushInput = () => e.target.dispatchEvent(new Event("blur", { bubbles: true }));
+
+      const currentTime = document.timeline.currentTime;
+      if (!inputMap.has(e.target)) inputMap.set(e.target, { time: null, steps: -1 });
+      inputMap.get(e.target).time = currentTime;
+      const newNumOfSteps = (inputMap.get(e.target).steps += 1);
+
+      if (newNumOfSteps === FLUSH_AFTER_N_STEPS) {
+        flushInput();
+        inputMap.delete(e.target);
+      } else {
+        setTimeout(() => {
+          if (inputMap.get(e.target).time === currentTime) {
```

Comment:
`time` is sometimes undefined here.

Reply:
Right, forgot to change this after I made it so entries are deleted.

`?.` would probably fix this, but I'd like to check